### PR TITLE
Disjoint Capture Diagnostics

### PR DESCRIPTION
**Pull request by ChrisPardy**

### CODE REVIEW

File: compiler/rustc_mir/src/borrow_check/diagnostics/move_errors.rs [hunk 1]
```diff
@@ -478,8 +478,11 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
                 self.note_type_does_not_implement_copy(err, &place_desc, place_ty, Some(span), "");
 
                 use_spans.args_span_label(err, format!("move out of {} occurs here", place_desc));
-                use_spans
-                    .var_span_label(err, format!("move occurs due to use{}", use_spans.describe()));
+                use_spans.var_span_label(
+                    err,
+                    format!("move occurs due to use{}", use_spans.describe()),
+                    "moved",
```
Comment (arora-aman) [inline compiler/rustc_mir/src/borrow_check/diagnostics/move_errors.rs:482]: should this not be use_spans.describe()?

File: compiler/rustc_mir/src/borrow_check/diagnostics/move_errors.rs [hunk 2]
```diff
@@ -348,7 +348,7 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
                 // FIXME(project-rfc-2229#8): Improve borrow-check diagnostics in case of precise
```
Comment (arora-aman) [inline compiler/rustc_mir/src/borrow_check/diagnostics/move_errors.rs]: remove the FIXME

File: compiler/rustc_mir/src/borrow_check/diagnostics/mod.rs [hunk 3]
```diff
@@ -630,14 +644,35 @@ impl UseSpans<'_> {
         }
     }
 
+    // Add a span label to the use of the captured variable, if it exists.
+    // only adds label to the `path_span`
+    pub(super) fn var_span_label_path_only(
```
Comment (arora-aman) [inline compiler/rustc_mir/src/borrow_check/diagnostics/mod.rs:657]: Why separate this out?

Comment (ChrisPardy) [reply compiler/rustc_mir/src/borrow_check/diagnostics/mod.rs:657]: We don't always need both pieces of information. e.g. 
```rust
fn main() {
    let mut c = {
        let mut p = Point {x: "1".to_string(), y: "2".to_string() };
        || {
           let x = &mut p.x;
           println!("{:?}", p);
            //~^ ERROR `p` does not live long enough
        }
    };
    c();
}
```
Here, we really only care that p is borrowed, not that it is a mutable borrow or why it is mutable, so just showing path span should be fine. This shuold allow for better signal to noise ratio in some messages.

File: compiler/rustc_mir/src/borrow_check/diagnostics/move_errors.rs (hunk 2 above)
Comment (ChrisPardy) [reply compiler/rustc_mir/src/borrow_check/diagnostics/move_errors.rs]: done

File: compiler/rustc_mir/src/borrow_check/diagnostics/mod.rs (hunk 3 above)
Comment (arora-aman) [reply compiler/rustc_mir/src/borrow_check/diagnostics/mod.rs:657]: I see

**Comment by arora-aman** [comment]
LGTM

**Pull request merged**